src/libsyntax/ext/tt/macro_check.rs

```diff
@@ -0,0 +1,621 @@
+//! Checks that meta-variables in macro definition are correctly declared and used.
+//!
+//! # What is checked
+//!
+//! ## Meta-variables must not be bound twice
+//!
+//! ```
+//! macro_rules! foo { ($x:tt $x:tt) => { $x }; }
+//! ```
+//!
+//! This check is also done at macro instantiation.
```

Comment:
Left-hand side checks cannot give false-positives/false-negatives, right?
And this seems to be the only left-hand side check, and it already produces a hard error.
So, it seems like the instantiation-time error can be removed now.

Reply:
Oops. My comment was wrong. This check is not done at macro instantiation. It was already done at macro definition and I just moved it to this file. I updated the comment.